mindsay_sdk/client.py

```diff
@@ -40,7 +41,28 @@ def __init__(self, email: str, production: bool = False):
         self._user_sign_in(email)
         logger.warning('Connected to Mindsay on %s', environment)
 
-    def _user_sign_in(self, email: str):
+    @classmethod
+    def setup_bos_environment(cls, user_email: str, instance_name: Optional[int], language: Optional[str],
```

Comment:
BOS is a deprecated name and should not appear on public document/code.
Also, as this function works as a factory, I would rename it `create_client` or something similar.

From an UX point of view, I suggest `instance_name` and `language` to be mandatory parameters (thus not Optional). Setting instance and language is the first thing done in most scripts I've seen.

Also, as experiment_id might never be used for the moment, I would add a `None` default value to make it optional.